draft-nro-bulk-rdap.md

```diff
@@ -48,184 +48,294 @@ feature of this service.
-top-level "rdapConformance" member. But, when in conflict, the Bulk RDAP extension requirements SHOULD supersede the NRO
-RDAP Profile extension requirements, primarily to afford bulk data compactness.
+The "rdapConformance" array for each returned data object MUST include both the "nroBulkRdap1" extension identifier and
+the extension identifier for the
+[NRO RDAP Profile](https://bitbucket.org/nroecg/nro-rdap-profile/raw/v1/nro-rdap-profile.txt) ("nro_rdap_profile_0" as
+of this writing). In the data objects, the Bulk RDAP extension requirements MUST supersede the NRO RDAP Profile
+extension requirements, primarily to afford bulk data compactness.
 
 # Bulk RDAP URL {#bulk_rdap_url}
 
 * Scheme: HTTPS
 * Method: GET
 * Path Segment: nroBulkRdap1?objectClass=<RDAP object class name>
-* Returns: Bulk data as either JSON data ((#bulk_data_format)) or JSON Web Signature (JWS) string
-  ((#security_considerations))
-* Content-Type: application/rdap+json
+* Returns: See (#content_types)
+
+The HTTPS scheme MUST be used for the Bulk RDAP URL.
 
-Limiting the query to a single RDAP object class in the Bulk RDAP URL should result in a simpler service implementation,
-as well as better performance when serving requests.
+When the "objectClass" query parameter is present in the request URL, the bulk data response MUST be limited to all the
+objects for the RDAP object class listed in that query parameter.
 
-Here is an example URL to get bulk data for the IP Network object class:
+When the "objectClass" query parameter is absent in the request URL, the bulk data response MUST return all the objects
+for all the RDAP object classes the server supports.
+
+The following is an example URL to get bulk data for the IP Network object class:
 
 ```
 https://example.net/nroBulkRdap1?objectClass=ip%20network
 ```
 
-_Figure 2_
-
-For a 200 OK response ([@!RFC9110, section 15.3.1]), the server MUST return a JSON object with its "objects" member
-filled with all the objects for the RDAP object class listed in the "objectClass" query parameter.
-
-If the RDAP object class listed in the "objectClass" query parameter is valid ([@!RFC9083, section 5], or a future RDAP
+If the RDAP object class listed in the "objectClass" query parameter is valid ([@!RFC9083, section 5] or a future RDAP
 object class) but the bulk data functionality has not been implemented for it, the server SHOULD return a 501 Not
 Implemented response ([@!RFC9110, section 15.6.2]).
 
 If the RDAP object class listed in the "objectClass" query parameter is invalid, the server SHOULD return a 400 Bad
 Request response ([@!RFC9110, section 15.5.1]).
 
+The following is an example URL to get bulk data for all the RDAP object classes an RIR supports:
+
+```
+https://example.net/nroBulkRdap1
+```
+
+## Content Types {#content_types}
+
+The content type for a bulk data response can be:
+
+* "application/json" -- for a newline-delimited sequence of metadata JSON object and RDAP objects, or for JWS JSON
```

Comment:
Is there any prior art for using `application/json` for a JSONL-type response?  It seems counterintuitive for such a response not to be parseable with a standard JSON parser.

Reply:
That's an excellent question. What I could gather is that the JSONL implementors "loosely" use `application/json`, say for bulk data in the form of JSON log files. But, to be technically correct, I would say `text/plain` is better, no?

Reply:
I would prefer a specific content type be registered for JSONL, if it's going to be used.  (If it can't happen in this document for whatever reason, then I'd lean towards JSON text sequences to save the trouble of having to spin up another document just for that content type.)

Reply:
Yah, my heart says: "Go with JSONL because it is simpler than JSON Text Sequences but then you'd have to go through with (the trouble of :)) registering `application/jsonl`". But, my mind says: "Though JSON Text Sequences will be more onerous implementation-wise, and perhaps problematic because of that RS control character, go with it because you get already standardized `application/json-seq`." I'm also now leaning towards what my mind is saying -- JSON Text Sequences. (Mahesh, you were right all this time about using JSON Text Sequences. :))